InvenTree/InvenTree/mixins.py

```diff
@@ -81,10 +81,56 @@ class RetrieveUpdateAPI(CleanMixin, generics.RetrieveUpdateAPIView):
     """View for retrieve and update API."""
     pass
 
+# The CustomDestroyModelMixin, CustomRetrieveUpdateDestroyAPIView, CustomRetrieveUpdateDestroyAPI
+# classes were created to being able to pass the kwargs from the API to the models
+class CustomDestroyModelMixin:
+    """
+    Destroy a model instance.
+    """
+    def delete(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance, **kwargs)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def perform_destroy(self, instance, **kwargs):
+        instance.delete(**kwargs)
+
+class CustomDestroyModelMixin:
+    """
+    Destroy a model instance.
+    """
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance, **kwargs)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def perform_destroy(self, instance, **kwargs):
+        instance.delete(**kwargs)
```

Comment:
Question: Why is this class defined twice?

Reply:
Ctrl-C Ctrl-V Ctrl-V

Reply:
> Ctrl-C Ctrl-V Ctrl-V

Yes indeed :)